Issue #3005411: Default users for testing

mtodor opened the pull request:
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/develop/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

mtodor committed:
Added users for every provided role

dbosen commented on modules/thunder_demo/content/user/09ec113d-32a1-4dcf-80d9-fc78069ce7b1.json:
```diff
@@ -0,0 +1,101 @@
+    ],
+    "name": [
+        {
+            "value": "kayne"
```
I guess this should be Kanye :) - Well, actually it is just Ye now (https://en.wikipedia.org/wiki/Kanye_West#Name_%22Ye%22).

mtodor replied:
Ticket is defined on 24.09.2018. It as legacy naming. 🤣

dbosen commented on modules/thunder_demo/content/user/09ec113d-32a1-4dcf-80d9-fc78069ce7b1.json:
```diff
@@ -0,0 +1,101 @@
+        {
+            "value": {
+                "title": "kayne | Site-Install",
+                "canonical_url": "http:\/\/default\/user\/4",
```
This is usually wrong, can we skip the metatag information? The node exports just contain the field_meta_tags values.

dbosen commented on modules/thunder_demo/content/user/09ec113d-32a1-4dcf-80d9-fc78069ce7b1.json:
```diff
@@ -0,0 +1,101 @@
+    "default_langcode": [
+        {
+            "value": true,
+            "lang": "en"
```
Lang codes have to be removed everywhere, see node export.

dbosen reviewed (changes requested):
I am not sure about this. I did not know, that so many user should be generated, with all the funny names :).
In the distribution we should just have one user per role with simple technical names. Editor, restricted-editor, seo. 
The more complex setups like this should be in the testing only module.

mtodor committed:
Merge branch '8.x-2.x' into feature/3005411-add-default-users

mtodor committed:
Adjusted review comments

mtodor commented:
@dbosen I have addressed all the comments from the review. I've started work on a custom module with all listed testing users (btw. we need repo for that). And here we have one user per provided role. I would keep prefix `test-` before role to avoid possible conflicts with created `admin` or `administrator` user during installation process.

chrfritsch commented:
Could we open the context of this PR a bit and also add some tags, that are connected to our articles?

That would be helpful for normal testing and also to see some differences on https://github.com/BurdaMagazinOrg/theme-thunder-admin/pull/166

mtodor committed:
Merge branch '8.x-2.x' into feature/3005411-add-default-users

mtodor commented:
@chrfritsch - I would create new issue for adding of Tags for articles and make it in another PR. It's better to keep scopes smaller.

chrfritsch reviewed (dismissed):
Please, then make it so.

We also need a follow-up to set the new users as authors of our nodes.

mtodor committed:
Merge branch '8.x-2.x' into feature/3005411-add-default-users

mtodor commented:
Added issue for tags: https://www.drupal.org/project/thunder/issues/3008589
And for users assignment: https://www.drupal.org/project/thunder/issues/3008594

mtodor committed:
Merge branch '8.x-2.x' into feature/3005411-add-default-users
